### ISSUE TITLE
Balance Donate/Receive NOT working

### ISSUE THREAD

**Issue by TwelveExit** [xbebenk/MBR_xbebenkMod#155]
hi , before anything I would like to thank you for this great work.

I have noticed the Balance Donate/Receive ratio option located in "Schedule Donations" tab has not been working for the past 2 months.
When the bot opens profile page, the OCR can't detect the values related to the donate and receive and it gives the following error:
"Profile Report
Opening Profile page to read Attacks, Defenses, Donations and Recieved Troops
Profile window doesnt exist, exiting..."

I haven't mentioned this error earlier cuz it wasn't a priority, but I would appreciate it if you could fix it.
thanks again.

**Comment by xbebenk** [comment]
![image](https://github.com/xbebenk/MBR_xbebenkMod/assets/6594360/b5b51e79-c426-4d74-916d-6977f44d72b7)

**Comment by TwelveExit** [comment]
yes, profile report is working now. The balance donate/receive ratio is not being applied though. I have tested different ratios 1/2 or 1/3 and ..... regardless of what I set the ratio in bot, it ignores it and starts donating.

for example: My account has 2000 donates and 300 receives and bot is set to 1/3 ratio but still bot is donating troops.

**Comment by xbebenk** [comment]
and log says?

**Comment by TwelveExit** [comment]
I tried to recreate the issue, it was a hit and miss where sometimes the bot would follow the donate/receive ratio and sometimes it ignored it.

After couple hours, this is what I found, once the bot detects all storages are full which I have set in "Halt Attack if all storages full", it starts ignoring the "Balance donate/receive". So if the bot is maxed on resources it will start donating troops regardless of what I set the ratio, which I think is expected of bot? I'm not sure if this was your intention or accidental.

But if the account doesn't have full storages then it would donate according to donate/receive ratio.
I would consider this issue to be solved now,
thanks
